### PR TITLE
use \r as new line delimiter

### PR DESCRIPTION
**Pull request by pelikhan**
Fix for https://github.com/microsoft/pxt-microbit/issues/3003

### CODE REVIEW

File: libs/core/serial.ts [hunk 1]
```diff
@@ -102,7 +102,7 @@ namespace serial {
         // (memory) efficient than the C++ implementation, because the
         // strings are statically allocated and take no RAM
         switch (del) {
-            case Delimiters.NewLine: return "\n"
+            case Delimiters.NewLine: return "\r"
```
Comment (mmoskal) [inline libs/core/serial.ts:105]: I think it would be better to add Delimiters.Enter or Return or  CarriageReturn. There may be already people using that and passing data in from files.

Comment (mmoskal) [reply libs/core/serial.ts:105]: We should also fix the docs.

Comment (sjmurdoch) [reply libs/core/serial.ts:105]: I would also be interested to know what is the behaviour on platforms (e.g. TeraTerm on Windows, which I can't test). I was a bit surprised to find an issue in code that's been present for years, which makes me suspect it works for other people. Perhaps Windows does something different from MacOS here, and changing the delimiter to \r could break other platforms. I couldn't find any specification for what is the right behaviour of serial terminal emulators and suspect there's wide variation.

Comment (pelikhan) [reply libs/core/serial.ts]: To follow @mmoskal  suggestion, I have left the existing behavior unchanged, updated the docs and added a carriage return delimiter option. Since this behavior has been in effect for years, changing might break an unknown set of existing programs.

Comment (mmoskal) [reply libs/core/serial.ts]: I think you can often change it in the settings of the terminal, what it will send when you press enter. I think a single `\r` is quite common though.

File: libs/core/_locales/core-jsdoc-strings.json [hunk 2]
```diff
@@ -558,6 +558,7 @@
   "pins.pulseIn": "Return the duration of a pulse at a pin in microseconds.",
   "pins.pulseIn|param|name": "the pin which measures the pulse, eg: DigitalPin.P0",
   "pins.pulseIn|param|value": "the value of the pulse, eg: PulseValue.High",
+  "pins.pushButton": "Mounts a push button on the given pin",
```
Comment (sjmurdoch) [inline libs/core/_locales/core-jsdoc-strings.json:561]: Is this relevant to the commit?

File: libs/core/_locales/core-strings.json [hunk 3]
```diff
@@ -62,12 +62,15 @@
...
+  "Delimiters.CarriageReturn|block": "carriage return (\r)",
   "Delimiters.Colon|block": ":",
   "Delimiters.Comma|block": ",",
   "Delimiters.Dollar|block": "$",
   "Delimiters.Fullstop|block": ".",
   "Delimiters.Hash|block": "#",
-  "Delimiters.NewLine|block": "new line",
+  "Delimiters.NewLine|block": "new line (\n)",
+  "Delimiters.Pipe|block": "|",
+  "Delimiters.SemiColumn|block": ";",
```
Comment (sjmurdoch) [inline libs/core/_locales/core-strings.json:73]: Should be SemiColon (not SemiColumn)

File: libs/core/serial.ts [hunk 4]
```diff
@@ -1,3 +1,24 @@
...
+    //% block="."
+    Fullstop = 5,
+    //% block="#"
+    Hash = 6,
+    //% block="carriage return (\r)"
+    CarriageReturn = 7,
+    //% block="|"
+    Pipe = 8,
+    //% block=";"
+    SemiColumn = 9,
```
Comment (sjmurdoch) [inline libs/core/serial.ts:19]: Should be SemiColon (not SemiColumn)

File: libs/core/serial.ts [hunk 5]
```diff
@@ -108,6 +129,9 @@ namespace serial {
             case Delimiters.Colon: return ":"
             case Delimiters.Fullstop: return "."
             case Delimiters.Hash: return "#"
+            case Delimiters.CarriageReturn: return "\r"
+            case Delimiters.Pipe: return "|"
+            case Delimiters.SemiColumn: return ";"
```
Comment (sjmurdoch) [inline libs/core/serial.ts:134]: Should be SemiColon (not SemiColumn)

File: libs/core/_locales/core-jsdoc-strings.json (hunk 2 above)
Comment (pelikhan) [reply libs/core/_locales/core-jsdoc-strings.json:561]: No. It's a string from another change ; that hasn't build into the strings. ignore.

File: libs/core/serial.ts [hunk 6]
```diff
@@ -1,3 +1,24 @@
+const enum Delimiters {
+    //% block="new line (\n)"
+    NewLine = 1,
```
Comment (mmoskal) [inline libs/core/serial.ts:3]: maybe we could do NewLine = 10, Comma = 44, etc. and then have:
```
export function delimiters(del: Delimiters): string { return String.fromCharCode(del as number) }
```
This way you can always cast any number to `Delimiters` if you need to.

Comment (pelikhan) [reply libs/core/serial.ts:3]: I guess we have that function now

Comment (mmoskal) [reply libs/core/serial.ts:3]: (the memory efficiency comment on that function no longer applies due to MSTR())

Comment (pelikhan) [reply libs/core/serial.ts]: The delimiters function is mostly a helper for blocks, the API takes a string already.
